### PR TITLE
24 create gateway messages that allow bookmarks to work server side

### PR DESCRIPTION
**Pull request by dylanratcliffe**
Adds the new message types

**Breaking changes**

### CODE REVIEW

**Comment by dylanratcliffe** [comment]
@getinnocuous It's failing due to linting saying "Forgot to run Prettier?". Is that something I should have done? Is it documented?

Also the tests pass here but fail on my machine with:

```
> @overmindtech/sdp-js@1.0.1 test
> jest

ts-jest[config] (WARN) message TS151001: If you have issues related to imports, you should consider setting `esModuleInterop` to `true` in your TypeScript configuration file (usually `tsconfig.json`). See https://blogs.msdn.microsoft.com/typescript/2018/01/31/announcing-typescript-2-7/#easier-ecmascript-module-interoperability for more information.
 FAIL  src/__tests__/util.test.ts (19.612 s)
  ● Util namespace › #getHash() › should work for items

    ReferenceError: crypto is not defined

      263 |   const name = getGloballyuniquename(object)
      264 |   const textAsBuffer = new TextEncoder().encode(name)
    > 265 |   const hashBuffer = await crypto.subtle.digest('SHA-256', textAsBuffer)
          |                      ^
      266 |   const bytes = new Uint8Array(hashBuffer)
      267 |   const base32String = base32EncodeCustom(bytes)
      268 |

      at src/Util.ts:265:22
      at src/Util.ts:8:71
      at Object.<anonymous>.__awaiter (src/Util.ts:4:12)
      at getHash (src/Util.ts:176:12)
      at src/__tests__/util.test.ts:79:33
      at src/__tests__/util.test.ts:12:71
      at Object.<anonymous>.__awaiter (src/__tests__/util.test.ts:8:12)
      at Object.<anonymous> (src/__tests__/util.test.ts:78:44)

  ● Util namespace › #getHash() › should work for references

    ReferenceError: crypto is not defined

      263 |   const name = getGloballyuniquename(object)
      264 |   const textAsBuffer = new TextEncoder().encode(name)
    > 265 |   const hashBuffer = await crypto.subtle.digest('SHA-256', textAsBuffer)
          |                      ^
      266 |   const bytes = new Uint8Array(hashBuffer)
      267 |   const base32String = base32EncodeCustom(bytes)
      268 |

      at src/Util.ts:265:22
      at src/Util.ts:8:71
      at Object.<anonymous>.__awaiter (src/Util.ts:4:12)
      at getHash (src/Util.ts:176:12)
      at src/__tests__/util.test.ts:85:33
      at src/__tests__/util.test.ts:12:71
      at Object.<anonymous>.__awaiter (src/__tests__/util.test.ts:8:12)
      at Object.<anonymous> (src/__tests__/util.test.ts:83:49)

ts-jest[config] (WARN) message TS151001: If you have issues related to imports, you should consider setting `esModuleInterop` to `true` in your TypeScript configuration file (usually `tsconfig.json`). See https://blogs.msdn.microsoft.com/typescript/2018/01/31/announcing-typescript-2-7/#easier-ecmascript-module-interoperability for more information.
ts-jest[config] (WARN) message TS151001: If you have issues related to imports, you should consider setting `esModuleInterop` to `true` in your TypeScript configuration file (usually `tsconfig.json`). See https://blogs.msdn.microsoft.com/typescript/2018/01/31/announcing-typescript-2-7/#easier-ecmascript-module-interoperability for more information.
ts-jest[config] (WARN) message TS151001: If you have issues related to imports, you should consider setting `esModuleInterop` to `true` in your TypeScript configuration file (usually `tsconfig.json`). See https://blogs.msdn.microsoft.com/typescript/2018/01/31/announcing-typescript-2-7/#easier-ecmascript-module-interoperability for more information.
 PASS  src/__tests__/progress.test.ts (18.997 s)
 PASS  src/__tests__/autocomplete.test.ts (20.157 s)
 PASS  src/__tests__/gateway.test.ts (19.845 s)

Test Suites: 1 failed, 3 passed, 4 total
Tests:       2 failed, 98 passed, 100 total
Snapshots:   0 total
Time:        34.365 s
```

Any ideas?

**Comment by getinnocuous** [comment]
Have you done a fresh `npm install`? Theres been some new deps added since you last worked on it possibly

**Review by getinnocuous — COMMENTED** [review]
A few incredibly minor comments but overall looks good to me!

File: src/GatewaySession.ts [hunk 1]
```diff
@@ -130,6 +134,36 @@ export class GatewaySession extends EventTarget {
           })
         )
       }
+    } else if (response.hasDeleteitem()) {
+      const ref = response.getDeleteitem()
+
```
[the remark this answers is not in the document]
Comment (dylanratcliffe) [reply src/GatewaySession.ts:139]: It's part of the way the code generator works annoyingly, only camelCases the first letter :(

**Review by getinnocuous — APPROVED** [review]

**Pull request merged**